tapir/coop/emails/membershipresignation_transfered_shares_confirmation.py

```diff
@@ -0,0 +1,56 @@
+import os
+from typing import List
+
+from tapir import settings
+
+from tapir.coop.models import ShareOwner, MembershipResignation
+from tapir.core.tapir_email_base import TapirEmailBase
+from django.utils.translation import gettext_lazy as _
+
+
+class MembershipResignationTransferedSharesConfirmation(TapirEmailBase):
+    def __init__(self, resigned_member: MembershipResignation):
+        super().__init__()
+        self.resigned_member = resigned_member
+
+    @classmethod
+    def get_unique_id(cls) -> str:
+        return "tapir.coop.transfered_shares_confirmation"
+
+    @classmethod
+    def get_name(cls) -> str:
+        return _("Confirmation Email for transfered shares.")
+
+    @classmethod
+    def get_description(cls) -> str:
+        return _(
+            "Automatically sent to the member who received shares after a resignation."
+        )
+
+    def get_subject_templates(self) -> List:
+        return [
+            "coop/email/membershipresignation_transfered_shares_confirmation_subject.html"
+        ]
+
+    def get_body_templates(self) -> List:
+        return [
+            "coop/email/membershipresignation_transfered_shares_confirmation_body.html"
+        ]
+
+    def get_extra_context(self) -> dict:
+        return {
+            "resigned_member": self.resigned_member.share_owner,
+            "receiving_member": self.resigned_member.transfering_shares_to,
+            "contact_email_address": settings.EMAIL_ADDRESS_MEMBER_OFFICE,
+        }
+
+    @classmethod
+    def get_dummy_version(cls) -> TapirEmailBase | None:
+        share_owner = ShareOwner.objects.filter(user__isnull=False).order_by("?")[0]
+        mail = cls(resigned_member=share_owner)
```

Comment:
This is the source of your error
> tapir.coop.models.ShareOwner.share_owner.RelatedObjectDoesNotExist: ShareOwner has no share_owner.

A `ShareOwner` object is passed but a `MemberResignation` is expected. That would be more visible if we changed the parameter name from `resigned_member` to `member_resignation`.

The `get_dummy_version` function try to render the email as it would be if we sent it to an actual member with real data, just picked randomly. You could either:

- pick a random MemberResignation (`order_by("?")` randomizes the order so you get a random element) and send it to `cls(...)`. Just be aware that maybe there are no MemberResignation object, then you'll have to return None since we can't render the dummy version.
- Create a temporary MemberResignation object with random values. We can't use `MemberResignation.objects.create()`, but If you create it with `MemberResignation(...)` instead, you'll create the object in memory but not in the database, so that should be fine.

Reply:
I hope I did it right? Let's have a look at the new commit 🙏

Reply:
Looks good 👍 

This function is used for the [mail preview page](https://members.supercoop.de/core/email_list), in order to generate a fake but real-looking example.